Preference Access SSN Matching/REPWRITERSPECS/RB.PREFACCESS.SSNMATCHING

```diff
@@ -0,0 +1,1625 @@
+                    FORMAT("  LOC:999999",PREFACCESSLOC(PAFINDLOOP))
+     PREFACCESSSTATUS(PAFINDLOOP)=PASTATUSVALID
+     IF PREFACCESSENABLEDP(PAFINDLOOP)<>PARAMSETDPFLAG OR
+        PREFACCESSENABLEWD(PAFINDLOOP)<>PARAMSETWDFLAG OR
+        PREFACCESSENABLEIQ(PAFINDLOOP)<>PARAMSETIQFLAG THEN
+      DO
+       PREFACCESSSTATUS(PAFINDLOOP)=PASTATUSUPDATE
+       PREFACCESSSTATUSREASON(PAFINDLOOP)="Update Access Flags"
+       BNODEBUGLFLINE=BNODEBUGLFLINE+" PA ACCESS FLAGS NEED UPDATING"
+      END
+     CALL BNODEBUGOUT
+    END
+  END
+
+[* Check the list of PA records and look to see if there is already a
+** PA record which has unrestricted access to the same account either already
+** existing or going to be made (NEW)
+*]
+ IF PAFOUND=FALSE THEN
+  DO
+   FOR PAFINDLOOP=1 TO PACOUNT
+    DO
+     IF PREFACCESSIDTYPE(PAFINDLOOP)=2 AND
+        PREFACCESSACCT(PAFINDLOOP)=TMPPAACCT THEN
+      DO
+       UNRESTRICTEDFOUND=TRUE
+      END
+    END
+  END
+[* If a matching PA record was not found, then it needs to be created so
+** increment the PA count by 1 and add the new info with the new flag set
+*]
+ IF PAFOUND=FALSE AND
+    UNRESTRICTEDFOUND=FALSE AND
+    PACOUNT<PACOUNTMAX THEN
+  DO
+   PACOUNT=PACOUNT+1
+   PREFACCESSACCT(PACOUNT)=TMPPAACCT
+   PREFACCESSIDTYPE(PACOUNT)=TMPPAIDTYPE
+   IF TMPPAIDTYPE<=2 THEN
+    PREFACCESSID(PACOUNT)=TMPPAID
+   IF TMPPAIDTYPE=3 THEN
+    PREFACCESSEXTLOANACCTNUM(PACOUNT)=TMPEXTLOANACCT
+   PREFACCESSENABLEDP(PACOUNT)=PARAMSETDPFLAG
+   PREFACCESSENABLEWD(PACOUNT)=PARAMSETWDFLAG
+   PREFACCESSENABLEIQ(PACOUNT)=PARAMSETIQFLAG
+   PREFACCESSSTATUS(PACOUNT)=PASTATUSNEW
+   BNODEBUGLFLINE=FORMAT("    PA MATCH NOT FOUND - ADDED TO LIST AT INDEX 999",PACOUNT)
+   CALL BNODEBUGOUT
+  END
+END [PROCEDURE]
+
+PROCEDURE VALIDATEPALIST
+[* The list of Preference access records is cycled through. For each PA record
+** which has an ID type of 2 (unrestricted), the list is checked to see if there
+** are any other preference access records pointing to the same account but to a
+** specific share or loan. If found, the access record is marked for deletion
+*]
+ FOR PALOOP=1 TO PACOUNT
+  DO
+   IF PREFACCESSIDTYPE(PALOOP)=2 THEN [Only test PA record if it's unrestricted]
+    DO
+     FOR PACHECKLOOP=1 TO PACOUNT
+      DO
+       IF PACHECKLOOP<>PALOOP AND [Skip over the PA record currently being tested]
+          PREFACCESSACCT(PALOOP)=PREFACCESSACCT(PACHECKLOOP) THEN
+        DO
+         PREFACCESSSTATUS(PACHECKLOOP)=PASTATUSINVALID
+         PREFACCESSSTATUSREASON(PACHECKLOOP)="Unrestricted PA record exists"
+        END
+      END
+    END
+  END
+END [PROCEDURE]
+
+PROCEDURE BUILDPALIST
+[* Build a list of current Preference Access Records on the account.
+** Program assumes only one Preference record on the account.
+**
+** RETURN PACOUNT     Number of Preference Access Accounts
+*]
+ PACOUNT=0
+ PREFLOC=0
+ FOR EACH PREFERENCE
+  DO
```

Comment:
Should this loop have a PREFFOUND=TRUE statement to stop it from looping? Right now it will do this for all preferences on the account and just use the last one it finds. Just a suggestion as this could cause unneeded processing.

Reply:
The variable PREFLOC is tracking this.  PREFLOC is being set to 0 before the "FOR EACH" loop and then when the first Preference record is found, is being set to the locator value. Line 1046 (IF PREFLOC=0) stops any further Preference records from being processed
```
 PREFLOC=0
 FOR EACH PREFERENCE
  DO
   IF PREFLOC=0 THEN
    DO
     PREFLOC=PREFERENCE:LOCATOR
     BNOTEXT=FORMAT("######9",PREFLOC)

```